### ISSUE TITLE
SchemaFor.oneOf macros issue

### ISSUE THREAD

**Issue by tayvs** [softwaremill/tapir#152]
When using methods that return Boolean as Discriminator propertyName (field `extractor` in SchemaFor.oneOf)

Code example
```
    import tapir._
    import tapir.json.circe._
    import io.circe.generic.auto._

    sealed abstract class SummaryResponse(val isWaiting: Boolean)
    case class NotWaiting() extends SummaryResponse(false)
    case class WaitingForAnswer(playingTime: Long) extends SummaryResponse(true)

    SchemaFor.oneOf[SummaryResponse, Boolean](_.isWaiting, _.toString)(
      true -> implicitly[SchemaFor[WaitingForAnswer]],
      false -> implicitly[SchemaFor[NotWaiting]])
```

```
Exception in thread "main" java.lang.BootstrapMethodError: java.lang.NoSuchMethodError: packageName.Endpoints.$anonfun$$init$$3$adapted(Ljava/lang/Object;)Ljava/lang/String;
	at packageName.Endpoints$$anonfun$1.applyOrElse(Endpoints.scala:31)
	at packageName.Endpoints$$anonfun$1.applyOrElse(Endpoints.scala:31)
	at scala.PartialFunction.$anonfun$runWith$1$adapted(PartialFunction.scala:145)
	at scala.collection.immutable.Map$Map2.foreach(Map.scala:159)
	at scala.collection.TraversableLike.collect(TraversableLike.scala:274)
	at scala.collection.TraversableLike.collect$(TraversableLike.scala:272)
	at scala.collection.AbstractTraversable.collect(Traversable.scala:108)
	at packageName.Endpoints.$init$(Endpoints.scala:31)
	at packageName.AppEndpoint$.<init>(AppEndpoint.scala:6)
	at packageName.AppEndpoint$.<clinit>(AppEndpoint.scala)
	at packageName.AppEndpoint.main(AppEndpoint.scala)
Caused by: java.lang.NoSuchMethodError: packageName.Endpoints.$anonfun$$init$$3$adapted(Ljava/lang/Object;)Ljava/lang/String;
	at java.lang.invoke.MethodHandleNatives.resolve(Native Method)
	at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975)
	at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000)
	at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1394)
	at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1750)
	at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477)
	... 11 more
```

**Comment by adamw** [comment]
Hm this works fine for me. Which version/JVM?

**Comment by tayvs** [comment]
@adamw I'm just create new clear project without any not tapir dependencies and this error still reproducing. 
I use 1.8.0._191 jdk versoin, scala 2.12.8 and tapir versions 0.8.5 and today try on 0.8.6

**Comment by tayvs** [comment]
@adamw see https://github.com/tayvs/BugReproducingTapir It's mentioned project

**Comment by adamw** [comment]
Released in 0.8.7
